src/ucp/rndv/proto_rndv.c

```diff
@@ -1081,3 +1194,74 @@ void ucp_proto_rndv_bulk_request_init_lane_idx(
 
     req->send.multi_lane_idx = lane_idx - 1;
 }
+
+static int
+ucp_proto_rndv_predict_rtr_mtype(const ucp_proto_init_params_t *init_params)
+{
+    ucp_context_t *context          = init_params->worker->context;
+    ucs_memory_type_t frag_mem_type = context->config.ext.rndv_frag_mem_type;
+
+    if (init_params->rkey_config_key == NULL) {
+        return 0;
+    }
+
+    return (init_params->rkey_config_key->mem_type == frag_mem_type) ? 1 : 0;
```

Comment:
why is it considerd as mtype rtr in this case?

Reply:
Made condition more strict as it was discussed.